File: php/101-symmetric-tree.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($value) { $this->val = $value; }
 * }
 */
class Solution 
{
    /**
     * @param TreeNode $root
     * @return Boolean
     */
    public function isSymmetric($root) 
    {
        return $this->isSymmetricRecursive($root, $root);   
    }
    
    public function isSymmetricRecursive($left, $right) 
    {
        if ($left === null && $right === null) {
            return true;
        }
        if ($left && $right) {
            if ($left->val != $right->val) {
                return false;
            }
            return ($this->isSymmetricRecursive($left->left, $right->right) && $this->isSymmetricRecursive($left->right, $right->left));
        }
        return false;
    }
}

File: go/101-symmetric-tree.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
	return isSymmetricRecursive(root, root)
}
func isSymmetricRecursive(left, right *TreeNode) bool {
	if left == nil && right == nil {
		return true
	}
	if left == nil || right == nil {
		return false
	}
	if left.Val != right.Val {
		return false
	}
	return (isSymmetricRecursive(left.Left, right.Right)) && (isSymmetricRecursive(left.Right, right.Left))
}
